app/optscale_api/auth_api.py

```diff
@@ -34,6 +35,37 @@ class OptScaleAuth:
     def __init__(self):
         self.api_client = APIClient(base_url=settings.opt_scale_api_url)
 
+    async def validate_authorization(self, bearer_token: str, org_id: str):
```

Comment:
I would suggest either change the arguments of this function so you can validate the user is authorized for any actions (it means we should be able to pass all the three parameters (action, resource_type, uuid), either change the name of the function to `is_cloud_account_creation_allowed_for_user` or `can_create_cloud_accounts` or something similar.

For the returned value/exception I would suggest in the first case:

* Return None if the validation succeed, raise an exception if it fails
* Return the validation result so the called can check if the validation succeeded or not.

In the second case, a True/False return value is a better.